repos/fdbt-site/src/pages/api/capPricingPerDistance.ts

```diff
@@ -0,0 +1,120 @@
+import { startCase } from 'lodash';
+import { NextApiResponse } from 'next';
+import { CAP_PRICING_PER_DISTANCE_ATTRIBUTE } from '../../../src/constants/attributes';
+import { redirectTo } from '../../../src/utils/apiUtils';
+import { isCurrency, isValidNumber } from '../../../src/utils/apiUtils/validator';
+import { updateSessionAttribute } from '../../../src/utils/sessions';
+import { CapDistancePricing, DistanceCap, ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+
+export const checkInputIsValid = (inputtedValue: string | undefined, inputType: string): string => {
+    let error;
+
+    if (!inputtedValue) {
+        error = `${startCase(inputType)} cannot be empty`;
+    } else if (Math.sign(Number(inputtedValue)) === -1) {
+        error = 'This must be a positive number';
+    } else if (!isCurrency(inputtedValue)) {
+        error = 'This must be a valid price in pounds and pence';
+    }
+
+    if (error) {
+        return error;
+    }
+
+    return '';
+};
+
+export const validateInput = (
+    capPricePerDistances: CapDistancePricing[],
+    lastIndex: number,
+    minimumPrice: string,
+    maximumPrice: string,
+): ErrorInfo[] => {
+    const errors: ErrorInfo[] = [];
+    const minimumPriceError = checkInputIsValid(minimumPrice, 'Minimum price');
+    if (minimumPriceError) {
+        errors.push({
+            id: `minimum-price`,
+            errorMessage: minimumPriceError,
+        });
+    }
+    const maximumPriceError = checkInputIsValid(maximumPrice, 'Maximum price');
+    if (maximumPriceError) {
+        errors.push({
+            id: `maximum-price`,
+            errorMessage: maximumPriceError,
+        });
+    }
+    capPricePerDistances.forEach((cap, index: number) => {
+        const { distanceFrom, distanceTo, pricePerKm } = cap;
+
+        if (lastIndex !== index) {
+            if (!distanceTo || !isValidNumber(Number(distanceTo))) {
+                errors.push({
+                    id: `distance-to-${index}`,
+                    errorMessage: 'Distance to is required and needs to be number',
```

Comment:
This error shows when the user enters 4.33333 km - we need to be more specific in this case and tell them its 2 decimal places maximum